### ISSUE TITLE
Slight modification, large loss values?

### ISSUE THREAD

**Issue by grey-area** [shekkizh/FCN.tensorflow#38]
Thanks for your work here.

I'm trying to train a slightly modified* version of this network.

In the figure under the 'Observations' heading on the main github page for this repo (the image at logs/images/sparse_entropy.png') is the entropy on the y-axis the training or validation loss, with iterations on the x-axis? If so, I seem to be getting huge loss values in comparison. The plot has an entropy of 4.5 initially, decreasing to around 3.5 after 2500 iterations, whereas I'm getting an initial loss of 300-600, with training loss hovering between 30 and 100 by iteration 2500, and validation loss somewhere between 70 and 100 at iteration 2500.

Are these reasonable values, or has something gone seriously wrong here?

(*Details of the modified version of the network:
- I've reintroduced relu5_3 between conv5_3 and pool5
- I'm loading the parameters for the first two fully connected layers of VGG-19 from vlfeat.org/matconvnet/models/beta16/imagenet-vgg-verydeep-19.mat instead of initialising them randomly
- Due to memory constraints, I'm holding all parameters fixed apart from the last fully connected layer of VGG-19 and the deconvolution/upsampling layers. )

**Comment by shekkizh** [comment]
Hi,

The loss show on tensorboard is of training loss. 

> I'm getting an initial loss of 300-600, with training loss hovering between 30 and 100 by iteration 2500, and validation loss somewhere between 70 and 100 at iteration 2500

This seems about right with your changes.  Since you froze most of the layers the network has very few weights it can adjust and so the training might be slow. The loss will reduce further provided the network can fit it. I would keep training for a few more epochs and see how the loss varies.

Cheers.

**Comment by grey-area** [comment]
Thanks for your reply. I thought I'd report back because you or somebody else might find this interesting.

As I said, my reason for holding all parameters in VGG fixed apart from the last fully connected layer was due to memory constraints.

I've just modified my code to keep *only* the first fully connected layer fixed (the one that accounts for >70% of the parameters). In this case, I'm getting loss values of less than 4 within 40 iterations.

Even better, I'm able to increase the batch size from 2 to 8 without problems.

(I find it mysterious that this works. I thought holding parameters fixed saved memory because the gradients w.r.t. those parameters don't have to be stored. But if I'm holding a layer fixed while training layers preceding it, the gradients for the fixed layer are going to have to be stored at some point..?)
